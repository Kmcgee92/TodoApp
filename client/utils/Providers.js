import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
// provide redux and apollo
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

export default function ComponentWithProviders(
  ComponentToBeTested,
  props,
  initialState
) {
  let mocks = [];
  let store = mockStore(
    
    initialState || { auth: { activeUser: {}, error: false } }
  
  );
  return (
    <MockedProvider mocks={mocks}>
      <Provider store={store}>
        <ComponentToBeTested {...props} />
      </Provider>
    </MockedProvider>
  );
}
