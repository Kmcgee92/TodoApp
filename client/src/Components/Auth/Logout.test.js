import { mount } from "enzyme";
// component to test
import Logout from "./Logout";
// child components
import Button from "@material-ui/core/Button";
// utils
import Providers from "../../../utils/Providers";
describe("<Logout/>", () => {
  let props;
  let wrapWithProviders;
  let component;
  describe("Component Initial Rendering", () => {
    beforeEach(() => {
      props = {
        classes: {},
      };
      wrapWithProviders = mount(Providers(Logout, props));
      component = wrapWithProviders.find(Logout);
    });
    it("Should Render", () => {
      expect(component).toHaveLength(1);
    });
    it("Should Match Snapshot", () => {
      expect(component).toMatchSnapshot();
    });
    it("Should Render Cancel Button", () => {
      expect(component.find(Button)).toHaveLength(1);
    });
  });
  describe("Logout onClick Event", () => {
    beforeEach(() => {
      props = {
        handleLogout: jest.fn(),
        classes: {},
      };
      wrapWithProviders = mount(Providers(Logout, props));
      component = wrapWithProviders.find(Logout);
    });
    it("Should Trigger Callback on Button onClick", () => {
      expect(component.find(Button).props().onClick).toEqual(
        expect.any(Function)
      );
    });
  });
});
