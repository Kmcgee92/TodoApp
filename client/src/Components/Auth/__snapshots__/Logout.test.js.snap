// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout/> Component Initial Rendering Should Match Snapshot 1`] = `ReactWrapper {}`;
